OsmAnd/src/net/osmand/plus/track/cards/AuthorCard.java

```diff
@@ -0,0 +1,116 @@
+package net.osmand.plus.track.cards;
+
+import static net.osmand.aidlapi.OsmAndCustomizationConstants.CONTEXT_MENU_LINKS_ID;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.core.content.ContextCompat;
+
+import net.osmand.gpx.GPXFile;
+import net.osmand.gpx.GPXUtilities;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.R;
+import net.osmand.plus.activities.MapActivity;
+import net.osmand.plus.helpers.AndroidUiHelper;
+import net.osmand.plus.mapcontextmenu.other.ShareMenu;
+import net.osmand.plus.routepreparationmenu.cards.MapBaseCard;
+import net.osmand.plus.utils.AndroidUtils;
+import net.osmand.util.Algorithms;
+
+public class AuthorCard extends MapBaseCard {
+	public static final int NO_ICON = -1;
+
+	private final GPXFile gpxFile;
+	private final boolean nightMode;
```

Comment:
we have nightMode in base class, which is [initialized in constructor](https://github.com/osmandapp/Osmand/blob/6f4ba1a7e1025b35cf5b2ebda5e6dc4ea97a9635/OsmAnd/src/net/osmand/plus/routepreparationmenu/cards/BaseCard.java#L61). It is better to avoid duplicating class variables in subclasses